packages/app/src/app/overmind/namespaces/dashboard/actions.ts

```diff
@@ -2097,34 +2081,35 @@ export const removeBranchFromRepository = async (
   try {
     await effects.api.removeBranchFromRepository(owner, repoName, name);
 
-    // Manually remove the data from the state based on the
-    // current page. Then sync in the background.
-    if (page === 'repositories') {
-      const repository = dashboard.repositories?.[activeTeam]?.find(
-        r => r.repository.owner === owner && r.repository.name === repoName
-      );
-
-      if (repository) {
-        repository.branches = repository.branches.filter(b => b.id !== id);
-      }
-
-      actions.dashboard.getRepositoriesByTeam({
-        teamId: activeTeam,
-        bypassLoading: true,
-      });
+    // Clean the branch references from all possible locations in the state
+    // 1. Repository with branches
+    // 2. Contributiob branches
+    // 3. Recent branches
+    const key = getProjectUniqueKey({
+      teamId: activeTeam,
+      owner,
+      name: repoName,
+    });
+    const repository = dashboard.repositoriesWithBranches[key];
+
+    if (repository) {
+      dashboard.repositoriesWithBranches = {
+        ...dashboard.repositoriesWithBranches,
+        [key]: {
+          ...repository,
+          branches: repository.branches.filter(b => b.id !== id),
+        },
+      };
     }
 
-    if (page === 'my-contributions') {
-      dashboard.contributions =
-        dashboard.contributions?.filter(b => b.id !== id) ?? [];
+    dashboard.contributions =
+      dashboard.contributions?.filter(b => b.id !== id) ?? [];
 
-      actions.dashboard.getContributionBranches();
-    }
+    dashboard.sandboxes.RECENT_BRANCHES =
+      dashboard.sandboxes.RECENT_BRANCHES?.filter(b => b.id !== id) ?? [];
 
     if (page === 'recent') {
```

Comment:
This means that the branch is visible on the recent page?

Reply:
don't know if you're talking about the conditional, but it means the user is on the recent page right now, so a fetch is needed

Reply:
Sorry yes, I should remember that in the actual comment it doesn't highlight the selected line. And yeah I see now, it threw me off because `page` was deconstructed from `branch`.